pkg/agent/openflow/pipeline.go

```diff
@@ -1469,5 +1475,6 @@ func NewClient(bridgeName, mgmtAddr string, enableProxy, enableAntreaPolicy bool
 	c.ofEntryOperations = c
 	c.enableProxy = enableProxy
 	c.enableAntreaPolicy = enableAntreaPolicy
+	c.enableTLVMap = networkConfig.TrafficEncapMode.SupportsEncap() && networkConfig.TunnelType == ovsconfig.GeneveTunnel && features.DefaultFeatureGate.Enabled(features.Traceflow)
```

Comment:
I see that we have `enableProxy` and `enableAntreaPolicy` as parameters already. maybe we should have `enableTraceflow` there as well. Or maybe just use the feature gate directly and remove these parameters, maybe it will be easier to maintain? @tnqn

Reply:
I like to check feature gate to reduce func arguments. But guess still better to save the booleans to the client struct to be used more efficiently later.

Reply:
Another one - do we need to check "networkConfig.TrafficEncapMode.SupportsEncap() && networkConfig.TunnelType == ovsconfig.GeneveTunnel && features.DefaultFeatureGate.Enabled(features.Traceflow)" here, or we just fail at agent parameter validation, if Traceflow is enabled but encapMode and tunnelType are not valid?

Reply:
> Another one - do we need to check "networkConfig.TrafficEncapMode.SupportsEncap() && networkConfig.TunnelType == ovsconfig.GeneveTunnel && features.DefaultFeatureGate.Enabled(features.Traceflow)" here, or we just fail at agent parameter validation, if Traceflow is enabled but encapMode and tunnelType are not valid?

Yes, if "networkConfig.TrafficEncapMode.SupportsEncap() && networkConfig.TunnelType == ovsconfig.GeneveTunnel && features.DefaultFeatureGate.Enabled(features.Traceflow)" is true, Antrea inter-Node Traceflow is supported and Agent will set enableTLVMap to true and deploy tun_metadata0 related flows.
Antrea will only support intra-Node raceflow if SupportsEncap is false or TunnelType is not Geneve, which means not to deploy tun_metadata0 related flows.

Reply:
> A few comments to the commit message:
Fixed. Thanks.